frontend/src/Pages/SignupPage/SignupPage.jsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState, memo } from 'react'
+import { Grid, Typography, TextField, InputLabel, Checkbox, FormControlLabel, Button, Divider } from '@mui/material'
+import ChirpLogo from '../../Images/ChirpLogo.svg'
+import bird from '../../Images/bird.jpg'
+import LoginStyles from '../Login Page/Login.module.css'
+import apiClient from '../../Services/apiClient'
+import { useNavigate } from 'react-router-dom'
+import DEVELOPMENT_API_BASE_URL from '../../Services/constants'
+
+function SignupPage () {
+  const [userInfo, setUserInfo] = useState({ fName: '', lName: '', username: '', email: '', password: '', confirmPassword: '' })
+  const handleInput = (input) => {
+    const { name, value } = input.target
+    setUserInfo({
+      ...userInfo,
+      [name]: value
+    })
+  }
+  const navigate = useNavigate()
+  const handleNavigate = () => {
+    navigate('/dashboard')
+  }
+  const [error, setError] = useState({})
+  // error checking
+  function isValid () {
+    const errors = {}
+    if (!userInfo.fName) {
+      errors.fName = 'Enter the First Name.'
+    }
+    if (!userInfo.lName) {
+      errors.lName = 'Enter the Last Name.'
+    }
+    if (!userInfo.username) {
+      errors.username = 'Enter the Username.'
+    }
+    if (!userInfo.email) {
+      errors.email = 'Enter the Email'
+    }
+    if (!userInfo.password) {
+      errors.password = 'Enter the password'
+    } else if (userInfo.password !== userInfo.confirmPassword) {
+      errors.confirmPassword = "The password doesn't match"
+    }
+
+    setError(errors)
+    return Object.keys(errors).length === 0
+  };
+  const [checked, setChecked] = useState(false)
+  const isChecked = (event) => {
+    setChecked(event.target.checked)
+  }
+  const handleSignup = (event) => {
+    event.preventDefault() // cancel the default event behavior (browser refresh)
+    if (isValid() && checked) {
+      // if it's valid, request registrations to the server
+      apiClient.register(userInfo).then(res => {
+        if (res.data.success) {
+          // console.log('userId ', res.data.userId);
```

Comment:
Looks good! Just have to get the auth context provider PR approved, and make a hotfix to store that info in the user object in this area of code :)